Feature/new help system (#1321)
inviwo/inviwo: CHANGELOG.md

Here we document changes that affect the public API or changes that needs to be communicated to other developers. 

## 2022-08-11 InviwoSetupInfo

> The module setup info in a workspace file will not only save information about processors and modules used in the workspace.

Reviewer: "now" instead of "not only"? Otherwise, what else is stored
```suggestion
The module setup info in a workspace file will now only save information about processors and modules used in the workspace.
```

## 2022-07-11 Help
Alt/Option clicking a processor in the processor network will now show the help for the given processor.

## 2022-07-11 Append workspace

> The file manu gained an "Append" function to append a workspace into an existing workspace. Holding down Alt/Option in the file lists (Recent/Examples/Tests) will also append the workspace instead of opening it.

Reviewer: ```suggestion
An "Append" function was added to the File menu to append a workspace into an existing workspace. Holding down Alt/Option in the file lists (Recent/Examples/Tests) will also append the workspace instead of opening it.
```

Here we document changes that affect the public API or changes that needs to be communicated to other developers. 

## 2022-08-16 New help system
The old processor help system based on doxygen and qhp has been removed in favor of a runtime system where help text is added to processor/port/properties at runtime. To add a help text to a processor now you add a new element to the ProcessorInfo for that processor like this
```c++
const ProcessorInfo InstanceRenderer::processorInfo_{
"org.inviwo.InstanceRenderer",  // Class identifier
"Instance Renderer",            // Display name
"Mesh Rendering",               // Category
CodeState::Stable,              // Code state
Tags::GL,                       // Tags
R"(
    Renders multiple instances of a mesh.
    Each instance can be modified using uniform data provided 
    from a set of dynamic inports holding vectors of data. 

>     The the number of inport and types can be controlled using

Reviewer: ```suggestion
    The number of inports and types can be controlled using
```

    a List Property.
    
    The rendering will happen along these lines:
    
        for (auto uniforms : zip(dynamic vector data ports)) {
            for(uniform : uniforms) {
                shader.setUniform(uniform.name, uniform.value);
            }
            shader.draw();
        }
    
    How the uniforms are applied in the shader can be specified
    in a set of properties.
    
    Example network:
    [basegl/instance_renderer.inv](file:///<modulePath>/data/workspaces/instance_renderer.inv)
)"_unindentHelp};
```
The `_unindentHelp`  suffix will remove any leading indent and parse the string as a markdown document and then convert it to a inviwo document, which is what is expected by the `ProcessorInfo` struct. 
One can refer to images like so:
```c++
"![](file:///<modulePath>/docs/images/heightfield-network.png)"
```
or networks
```c++
"[basegl/instance_renderer.inv](file:///<modulePath>/data/workspaces/instance_renderer.inv)"
```
Images will be shown inline, and clicking on processor network will append them to the current network. 

> To be able to refer to files there are two placeholder `<basePath>` and `<modulePath>` the former will refer to the base path of the inviwo installation and the later to the current module. Make sure that any resources linked are also included in the installer. By default the `data` directory and the `docs` are always included in the installer. 

Reviewer: ```suggestion
To be able to refer to files there are two placeholder `<basePath>` and `<modulePath>` the former will refer to the base path of the inviwo installation and the latter to the current module. Make sure that any resources linked are also included in the installer. By default the `data` directory and the `docs` folders are always included in the installer. 
```

For Ports a second constructor argument has been added for a help Document like so
```cpp
inport_("mesh", "Mesh to be drawn multiple times"_help)
```

> the `_help` suffix will parse the string as markdown and convert it to a inviwo Document.

Reviewer: global nitpick:
search for "a inviwo", replace with "an inviwo"

For Properties a new constructor with a new third help argument after the identifier and display name has been added. For example the customInputDimensions in the CanvasProcessor:
```cpp
customInputDimensions_{"customInputDimensions",
                       "Image Size",
                       "The size of the image that will be generated. This can be larger "
                       "or smaller than the canvas size. A smaller size will generate a "
                       "blurrier canvas, but render faster. A larger size will render "
                       "slower."_help,
                       size2_t(256, 256),
                       {size2_t(1, 1), ConstraintBehavior::Immutable},
                       {size2_t(10000, 10000), ConstraintBehavior::Ignore},
                       size2_t(1, 1),
                       InvalidationLevel::Valid}
```

> The Help widget will automatically show the help text for the processor and combine the with the help text for the ports and properties in a standard way.  The help text for Ports and Properties will also be shown in relevant tooltips.

Reviewer: ```suggestion
The Help widget will automatically show the help text for the processor and combine it with the help text for the ports and properties in a standard way.  The help text for Ports and Properties will also be shown in relevant tooltips.
```
